saleor/graphql/core/utils/reordering.py

```diff
@@ -36,13 +37,14 @@ def __init__(self, qs: QuerySet, operations: dict[int, int], field: str):
         # Will contain the list of keys kept
         # in correct order in accordance to their sort order
         self.ordered_pks: list[int] = []
+        self.second_sort_field = second_sort_field
 
     @cached_property
     def ordered_node_map(self):
         ordering_map = OrderedDict(
             self.qs.select_for_update()
             .values_list("pk", "sort_order")
-            .order_by(F("sort_order").asc(nulls_last=True), "id")
+            .order_by(F("sort_order").asc(nulls_last=True), self.second_sort_field)
```

Comment:
```suggestion
            .order_by(F("sort_order").asc(nulls_last=True), self.second_sort_field, "id")
```
Maybe? The last ordering field should be unique to avoid breaking the pagination.

Reply:
@fowczarek Thanks, added :)